modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/resource/v1_0/TaxonomyVocabularyResourceImpl.java

```diff
@@ -120,6 +134,29 @@ public Page<TaxonomyVocabulary> getAssetLibraryTaxonomyVocabulariesPage(
 			assetLibraryId, search, filter, pagination, sorts);
 	}
 
+	@Override
+	public TaxonomyVocabulary
+			getAssetLibraryTaxonomyVocabularyByExternalReferenceCode(
+				Long assetLibraryId, String externalReferenceCode)
+		throws Exception {
+
+		AssetVocabulary assetVocabulary =
+			_assetVocabularyLocalService.
+				getAssetVocabularyByExternalReferenceCode(
+					assetLibraryId, externalReferenceCode);
+
+		String resourceName = getPermissionCheckerResourceName(
```

Comment:
Maybe we could refactor this code and the code from `getSiteTaxonomyVocabularyByExternalReferenceCode` and move it to a private method to avoid duplicated code

Reply:
Make sense, I will change it 😁